kolibri/core/assets/src/objectSpecs.js

```diff
@@ -75,10 +75,18 @@ function _validateObjectData(data, options, dataKey) {
 
   // object sub-spec
   if (hasData && options.spec) {
-    if (!isObject(data)) {
-      return _fail('Only objects can have sub-specs', dataKey, data);
+    if (!isObject(data) && !isArray(data)) {
+      return _fail('Only objects or arrays can have sub-specs', dataKey, data);
     }
-    if (!validateObject(data, options.spec)) {
+    // If it is an array, we will validate each item in the array
+    if (isArray(data)) {
+      for (const item of data) {
+        if (!validateObject(item, options.spec.spec)) {
+          return _fail('Object in Array sub-spec failed', dataKey, item);
+        }
+      }
+    }
+    if (isObject(data) && !isArray(data) && !validateObject(data, options.spec)) {
```

Comment:
There are some unreachable conditions here, considering the first `if` ensuring `data` is only an object or an array. Making this an `else if` would remove `!isArray(data)`, but also, would `isObject` and `isArray` ever be true at the same time?

Reply:
It's because `isObject` is super general and comes back `true` for an Array (and functions) -  so `isObject && !isArray` needs the `!isArray` check there because above that bit I'm checking for `isArray` and handling it when `true` but don't return so the last step is saying "this is an object that isn't an array" -- which probably should also check "is also not a function".

Although - now that I look at it again, it'd probably be simpler to just return after succeeding within the block where `isArray` is `true` and passes validation. Will push an update

Reply:
Since this is just for the object spec, perhaps `isPlainObject` is better? https://lodash.com/docs/4.17.15#isPlainObject

Reply:
Bah - nevermind I misread the GH diff - it can't return early.

I added a comment to clarify a bit.

Reply:
Awesome thanks for that -  all uses of `isObject` were really trying to ask `isPlainObject` so I replaced them throughout.